test.js

```diff
@@ -199,9 +234,34 @@ describe('capture-exit', function() {
       });
     });
 
+    it('throws an exit code of the first registered handler which rejects', function() {
+      exit.captureExit();
+
+      exit.onExit(handler({
+        timeout: 10
+      }));
+
+      exit.onExit(handler({
+        code: 400,
+        timeout: 20
+      }));
+
+      exit.onExit(handler({
+        code: 503
+      }));
+
+      return exit._flush()
+        .then(function() {
+          throw new Error('Promise should be rejected');
+        })
```

Comment:
This fulfillment handler is unnecessary: the handlers above are already rejecting the promise.